diff --git a/build-aux/config.guess b/build-aux/config.guess
index 7f76b62..9eab862 100755
--- a/build-aux/config.guess
+++ b/build-aux/config.guess
@@ -109,7 +109,7 @@ GUESS=
 
 tmp=
 # shellcheck disable=SC2172
-trap 'test -z "$tmp" || rm -fr "$tmp"' 0 1 2 13 15
+trap 'test -z "$tmp" || rm -fr "$tmp"' EXIT HUP INT PIPE TERM
 
 set_cc_for_build() {
     # prevent multiple calls if $tmp is already set
