diff --git a/gl/lib/fdopendir.c b/gl/lib/fdopendir.c
index bdbb2ea..270648f 100644
--- a/gl/lib/fdopendir.c
+++ b/gl/lib/fdopendir.c
@@ -138,7 +138,9 @@ fdopendir_with_dup (int fd, int older_dupfd, struct saved_cwd const *cwd)
         }
       else
         {
+#ifndef __MVS__
           close (fd);
+#endif
           dir = fd_clone_opendir (dupfd, cwd);
           saved_errno = errno;
           if (! dir)
